BEMIO Cleanup

akeeste opened the pull request:
This scripts completes the BEMIO cleanup project board task:
- [x] rename BEMIO variables and internal functions to camel case
- [x] rename BEMIO scripts to camel case
- [x] removed redundant writeh5 functions

updates after review:
- [x] changed ``writeH5`` to ``writeBEMIOH5`` to prevent confusion with matlab functions
- [x] updated ``H5CreateWriteAtt`` to ``writeH5Parameter`` since it creates a variable, writes values, and defines attributes
- [x] removed ``bemio.m`` and replaced it with a short ``README.md`` pointing to the examples/BEMIO cases
- [x] removed specific BEM names in ``normalizeBEM.m``, it is general now
- [x] removed the ``loadH5`` function since it replicates a lot of what the built in ``h5read`` does. The only purpose it served for us was to reverse the order of the dimensions upon reading from the h5 file. I'm not sure why this is necessary, but I created a short utility function that will does this same thing after using ``h5read``

updates building on Kelley's compareBEMIO PR into this fork:
- [x] test plotting for different number of WEC bodies, single/double/triple, e.g. sphere, rm3, wec3
- [x] test plotting for single, double and triple BEMIO results, e.g. WAMIT, Capytaine, AQWA
- [x] test plotting when nested in `plotBEMIO`, resolve issues with nested varargin functions
- [x] remove *.mat and AQWA files
- [x] add examples/**/*.png to gitignore so that images of BEMIO plots are not included on the repo 
- [x] Point bodyClass to the readBEMIOH5 function
- [x] refactor plotting scripts to remove the hydro parameter entirely. Also functions use ``varargin`` for simplicity
- [x] create new test for the compareBEMIO functionality by running `plotBEMIO` for two different Sphere BEM runs

Future Task --> project board
- [ ] Compare BEM data by loading h5 data into hydro data structure. Right now, this comparison is based on a temporary hydro structure from the BEM output files

kmruehl commented on source/functions/utils/v5Renaming.m:
```diff
@@ -0,0 +1,39 @@
+
+% Directory of the *.m files that will be updated
```
Please remove from PR but add to team board

akeeste replied:
Removed

kmruehl commented on source/functions/utils/updateNaming.m:
```diff
@@ -0,0 +1,50 @@
+function updateNaming(fileList,oldVariableName,newVariableName)
+% This file will read EVERY file in fileList and replace the
```
Please remove from PR but share with team

kmruehl commented on source/functions/BEMIO/writeH5/h5bemWriteBody.m:
```diff
@@ -0,0 +1,34 @@
+function h5bemWriteBody(filename, bodyNumber, cb, cg, dispVol, K, re, im, addedMass, addedMassInf, radiationDamping)
+	% Function to write a body's properties and coefficients to an 
+    %     existing WEC-Sim/BEMIO-formatted h5 file
```
I think this needs to be removed, and has been replaced by ``source/functions/BEMIO/writeH5.m``

kmruehl commented on source/functions/BEMIO/writeH5/h5bemWriteParameters.m:
```diff
@@ -1,20 +1,20 @@
-function h5bem_writeParameters(filename, wave_dir, water_depth, w, T)
+function h5bemWriteParameters(filename, waveDir, waterDepth, w, T)
 	% Function to write the simulation parameters to an existing 
     %     WEC-Sim/BEMIO-formatted h5 file
```
I think this needs to be removed, and has been replaced by ``source/functions/BEMIO/writeH5.m``

kmruehl commented on source/functions/BEMIO/writeH5/h5bemCreate.m:
```diff
@@ -1,4 +1,4 @@
-function h5bem_create(filename, numBodies, numFreq, numWaveDir, bodyNames, scaled)
+function h5bemCreate(filename, numBodies, numFreq, numWaveDir, bodyNames, scaled)
 	% Function to create a WEC-Sim/BEMIO-formatted h5 file and populate 
     %     the string datasets
```
I think this needs to be removed, and has been replaced by ``source/functions/BEMIO/writeH5.m``

kmruehl commented on source/functions/BEMIO/waveNumber.m:
```diff
@@ -63,31 +63,31 @@
     v = 0;  %current velocity in y direction [m/s]
 % end
 
```
Can we remove `waveNumber` from `waveClass` and only call on this? How do they differ?

akeeste replied:
This BEMIO functions contains a more complex iteration method that also considers wave current effects on the wavenumber. However this BEMIO function is only called by Read_NEMOH() for the mean drift calculations, and it never actually inputs current data. I replaced this with the waveClass function because it is more clear and should converge to the same value.

kmruehl reviewed (changes requested):
Thanks Adam, LMK if you have any questions

akeeste commented:
@kmruehl I made the requested changes, namely:
- removed renaming scripts. I can't find away to attach files to a team discussion thread, but have them locally if needed
- moved the ``waveClass.waveNumber`` method to the ``BEMIO/waveNumber.m`` function so that it's accessible by both ``Read_NEMOH`` and ``waveClass``
- removed redundant write H5 functions. These have been incorporated into ``writeH5.m`` previously

kmruehl commented:
- What's the difference between `h5read`  and `loadH5`? Both used in `bodyClass`
- Naming convention for `h5write` and `writeH5`, distinguish between built-in MATLAB and BEMIO functions
- internal functions, e.g. `H5CreateWriteAtt`
- What is purpose of `bemio.m`? Should we keep it and add description? Remove it and point to examples?
- `normalizeBEM` update description to include Capytaine or remove BEM names

kmruehl commented:
@akeeste  PR [#11](https://github.com/akeeste/WEC-Sim/pull/11) should resolved bug (unchecked boxes above) with the nested varargin. Should we add/update bemio tests for this?

akeeste commented:
@kmruehl I have finished the last clean up items on this PR. Everything seems to be working now and all tests pass locally. I went through all the files changed and don't see anything unintentional. I'm happy with merging this PR and then working on new tests/documentation separately.

kmruehl commented:
@akeeste I'll review this again and merge it. I can work on updating the tests, if you want to work on the documentation updates. We can work off the same PR though, I'll open one and tag you in it once this is merged.

kmruehl commented on source/functions/BEMIO/waveNumber.m:
```diff
@@ -1,125 +1,42 @@
+        k = w.^2./g./tanh(k.*waterDepth);
+    end
 end
 
```
Note to self to double check this formulation for shallow, intermediate and deep water

kmruehl replied:
Verified this is correct, and added some comments to clarify

kmruehl reviewed (commented):
Note to self, double check the waveNumber calculation for shallow, deep and intermediate water

kmruehl commented:
@akeeste thanks!

akeeste commented:
> @akeeste I'll review this again and merge it. I can work on updating the tests, if you want to work on the documentation updates. We can work off the same PR though, I'll open one and tag you in it once this is merged.

This sounds good. Wednesday before seeing this I started on reorganizing these tests to be more efficient. Let me know when you start and we could use that branch as a jumping off point.

kmruehl commented:
@akeeste I already updated the tests in this PR. However, I did not restructure them, I just added a test for the new plot feature. If you'd like to submit your restructured tests as a PR, feel free to.

akeeste commented:
@kmruehl Okay got it. I'll submit some restructuring when I update BEMIO documentation. I'm hoping to hold on to the wamit/nemoh/other hydrodata whenever it is read in to save time.
